lib/api.rb

```diff
@@ -59,6 +59,14 @@ def validation_error(msg)
     if !js.strip.blank?
       params.merge!(JSON.parse(js))
     end
+  ## needed for the web vault, which doesn't use the content-type  
+  elsif request.accept.to_s.match(/application\/json/)
+    js = request.body.read.to_s
+    if !request.content_type.to_s.match(/application\/x-www-form-urlencoded/)
+      if !js.strip.blank?
+        params.merge!(JSON.parse(js))
+      end
+    end
```

Comment:
This block should be merged into the one above it, to unify JSON reading and parsing (just checking `request.accept` vs `request.content_type`.

Reply:
Ok.